vm/src/vm.h

```diff
@@ -1,6 +1,10 @@
 #pragma once
 #include <cstddef>
 
+#if UINTPTR_MAX > 0xFFFFFFFFFFFFFFFFu // 64 bits
+#error "VM only supports architectures less than 64 bits architectures"
+#endif
+
```

Comment:
Why a larger address space would be an issue?

Reply:
because the bytecode defines a reference on 8 bytes (64 bits). which supports 64/32/16 etc architectures but cannot support larger addresses. We do not want to make the bytecode platform dependent and the VM has no function signatures to know the size of its stack/parameter/return value